packages/flutter_tools/lib/src/android/application_package.dart

```diff
@@ -103,11 +103,21 @@ class AndroidApk extends ApplicationPackage implements PrebuiltApplicationPackag
     required ProcessUtils processUtils,
     required Logger logger,
     required FileSystem fileSystem,
+    BuildInfo? buildInfo,
   }) async {
-    File apkFile;
+    final File apkFile;
```

Comment:
Rather than concatenating to the string, I think this is a little easier to read if the branches yield the complete string, like:

```dart
final File apkFile;
final String filename;
if (buildInfo == null) {
  fileName = 'app.apk';
} else if (buildInfo.flavor == null) {
  fileName = 'app-${buildInfo.mode.name}.apk';
} else {
  fileName = 'app-${buildInfo.lowerCasedFlavor}-${buildInfo.mode.name}.apk';
}
```

Reply:
That looks the same as implemented IMO. Could we leave another one for opinion? 😄

Reply:
This is easier to read, as it doesn't have nested branches, and it each final string is on a single line.

Reply:
I prefer the better readability in Chris's suggestion, as it is both more searchable and easier for people to comprehend in the future when revisiting this code.

Reply:
Thanks for these explanations, applying.